### DISCUSSION TITLE
How to not set replicas in Kafkaconnect

### DISCUSSION THREAD

**Discussion by ee07dazn** [strimzi/strimzi-kafka-operator#7380] [Q&A] [answered]
I am trying to use argocd for Gitops along with strimzi kafka connect. 
I want to also use HPA 
In order to maintain harmony between HPA and ArgoCD, I should not set `replicas` in Kafkaconnect deployment and let HPA handle it, without creating issues with ArgoCD. 

But from what i am noticing, if I comment out `replicas` from Kafkaconnect kind, it defaults to a value of 3. 

Is there a way to avoid setting the replicas in Deployment of Kafkaconnect? And let HPA handle it as also suggested [here](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#replicas)

**Comment by scholzj** [comment, marked as answer]
No, that is not the right approach. You always need to set the replicas in Kafka Connect. If you want to use autoscaling, you have to point your autoscaler to manage the replicas field in the KafkaConnect resource and not in the deployment. It has the `/scale` subresource required for it.

**Reply by ee07dazn** [reply]
🙌
